### PR TITLE
LFS tuple space tests

### LINKED ISSUE #3097: Additional tests for new options for LFS
## Overview
With new options to run the node for LFS (trim state, disable exporter) we need additional tests to cover different configurations.

### Impact
See PR #3079 and the changes to Engine instances.

### Design
Add additional tests with different combination of config parameters and assert that handling is correct.

### PR DESCRIPTION
**Pull request by tgrospic**
## Overview
This PR adds tests for tuple space part for Last Finalized State. It's continuation of PR #3160.

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: casper/src/main/scala/coop/rchain/casper/engine/LfsTupleSpaceRequester.scala [hunk 1]
```diff
@@ -149,83 +160,79 @@ object LfsTupleSpaceRequester {
...
-                      historyItemsBytes,
-                      dataItemsBytes,
-                      startPath,
-                      chunkSize = pageSize,
-                      skip = 0,
-                      importer.getHistoryItem
-                    )
-                  )
-                }
+        _ <- (Sync[F].defer[Unit] _ andThen Stream.eval) {
```
Comment (nzpr) [inline casper/src/main/scala/coop/rchain/casper/engine/LfsTupleSpaceRequester.scala:163]: what does this syntax mean?

**Review by nzpr — APPROVED** [review]
LGTM

File: casper/src/main/scala/coop/rchain/casper/engine/LfsTupleSpaceRequester.scala (hunk 1 above)
Comment (tgrospic) [reply casper/src/main/scala/coop/rchain/casper/engine/LfsTupleSpaceRequester.scala:163]: It's Scala way for function composition. It means `x => Stream.eval( Sync[F].defer[Unit]( x ) )`.

```scala
f andThen g   ==   x => g(f(x))   ==   g compose f
```

File: casper/src/test/scala/coop/rchain/casper/engine/LfsStateRequesterStateSpec.scala [hunk 2]
```diff
@@ -0,0 +1,113 @@
+package coop.rchain.casper.engine
+
+import coop.rchain.casper.engine.LfsTupleSpaceRequester.ST
+import org.scalatest.prop.GeneratorDrivenPropertyChecks
+import org.scalatest.{FlatSpec, Matchers}
+
+class LfsStateRequesterStateSpec extends FlatSpec with Matchers with GeneratorDrivenPropertyChecks {
+
+  "getNext" should "return empty list when called again" in {
+    val st = ST(Seq(10))
```
Comment (zsluedem) [inline casper/src/test/scala/coop/rchain/casper/engine/LfsStateRequesterStateSpec.scala:10]: The file here is actually testing the object `ST` which can be not related with LFS, I think. 
Maybe we can move this `ST` in shared folder.MAybe that can be reused in some other cases.

**Review by zsluedem — APPROVED** [review]
LGTM

File: casper/src/test/scala/coop/rchain/casper/engine/LfsStateRequesterStateSpec.scala (hunk 2 above)
Comment (tgrospic) [reply casper/src/test/scala/coop/rchain/casper/engine/LfsStateRequesterStateSpec.scala:10]: You are right. It was completely the same as in block requester but more logic is added for blocks. But for now better to create multiple variants of state object and see what stands out to be shared.
Maybe state can be a trait and different implementations can give us different features.

**Comment by tgrospic** [comment]
bors merge

**Pull request merged**